src/main/java/io/strimzi/kafka/bridge/Application.java

```diff
@@ -203,19 +217,53 @@ public static void main(String[] args) {
         return httpPromise.future();
     }
 
-    private static String getFilePath(String arg) {
-        String[] confFileArg = arg.split("=");
-        String path = "";
-        if (confFileArg[0].equals("--config-file")) {
-            if (confFileArg[1].startsWith(File.separator)) {
-                // absolute path
-                path = confFileArg[1];
-            } else {
-                // relative path
-                path = System.getProperty("user.dir") + File.separator + confFileArg[1];
-            }
-            return path;
+    /**
+     * Return a JmxCollectorRegistry instance with the YAML configuration filters provided
+     * through the command line or configured without any filters
+     *
+     * @param commandLine command line parameters containing the path to the JMX YAML configuration
+     * @return  JmxCollectorRegistry instanc
+     * @throws MalformedObjectNameException
+     * @throws IOException
+     */
+    private static JmxCollectorRegistry getJmxCollectorRegistry(CommandLine commandLine)
+            throws MalformedObjectNameException, IOException {
+        if (commandLine.hasOption("jmx-metrics-config")) {
+            return new JmxCollectorRegistry(new File(getFilePath(commandLine.getOptionValue("jmx-metrics-config"))));
```

Comment:
Does the file automatically reload like with the exporter?

Reply:
Yes, the underneath `JmxCollector` class does that. Every time you scrape for metrics, it checks if the configuration file is changed (checking the last modified time) and if needed, it reloads the configuration itself.